torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -152,7 +106,8 @@ void fuseBroadcast(std::shared_ptr<Graph>& graph) {
 
     n->replaceInput(input_index, unexpanded_rhs);
     n->i_(attr::broadcast, 1);
-    if (axis) {
+    // Gemm doesn't support the axis argument, so be sure to omit it for that op.
+    if (axis && n->kind() != onnx::Gemm) {
```

Comment:
What if the axis is not the default one?

Reply:
++ can enforce that it is